VDF.Core/Utils/GrayBytesUtils.cs

```diff
@@ -68,38 +67,121 @@ static class GrayBytesUtils {
 			return 100d / all * count >= darkProcent ? null : buffer;
 
 		}
+
 		[MethodImpl(MethodImplOptions.AggressiveInlining)]
 		public static unsafe float PercentageDifference(byte[] img1, byte[] img2) {
 			Debug.Assert(img1.Length == img2.Length, "Images must be of the same size");
-			long diff = 0;
-			if (Avx2.IsSupported) {
-				Vector256<ushort> vec = Vector256<ushort>.Zero;
-				Span<Vector256<byte>> vImg1 = MemoryMarshal.Cast<byte, Vector256<byte>>(img1);
-				Span<Vector256<byte>> vImg2 = MemoryMarshal.Cast<byte, Vector256<byte>>(img2);
 
-				for (int i = 0; i < vImg1.Length; i++)
-					vec = Avx2.Add(vec, Avx2.SumAbsoluteDifferences(vImg2[i], vImg1[i]));
+			if (Avx2.IsSupported && img1.Length % 32 == 0)
+				return PercentageDifferenceAvx2(img1, img2);
```

Comment:
Please inline the code again and not make it a call. Also why did you added % 32 == 0? This is always be the case.

Reply:
OK
_(Even if I found the division in principle nicer, because it has allowed to test also specifically the individual variants.)_

> added % 32 == 0
Because of my approach to make the width configurable. So as you requested not to do this, this is one of the  _"I'm not commenting other places where you changed it."_ 
As I had mentioned somewhere before, as I made `width` a parameter my focus was not to support the best performance for any other resolutions but not to break the functions if a unfavorable resolution would be selected.
And because of the 32 Byte MemoryMarshal mapping I had the feeling it could be a problem if the underlying Byte array is smaller then this region.